### PR TITLE
isescan: new package at 1.7.2.1

### PR DESCRIPTION
**Pull request by codeandkey**
depends on `fraggenescan` in #16587, `py-fastcluster` in #16588

### CODE REVIEW

File: var/spack/repos/builtin/packages/isescan/package.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    def setup_run_environment(self, env):
+        env.prepend_path('PATH', self.prefix)
+        env.prepend_path('LD_LIBRARY_PATH',
+                         join_path(self.prefix, 'ssw201507'))
+
+    def install(self, spec, prefix):
+        # build bundled SSW library
+        with working_dir('ssw201507'):
+            which('gcc')(
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/isescan/package.py]: Doesn't this only work for the GCC compiler? You should use `Executable(spack_cc)` to use the compiler wrappers.

File: var/spack/repos/builtin/packages/isescan/package.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                '-O3', '-pipe', '-fPIC', '-shared', '-rdynamic',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/isescan/package.py]:
```suggestion
                '-O3', '-pipe', self.compiler.cc_pic_flag, '-shared', '-rdynamic',
```

File: var/spack/repos/builtin/packages/isescan/package.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                '-o', 'libssw.so', 'ssw.c', 'ssw.h',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/isescan/package.py]:
```suggestion
                '-o', 'libssw.' + dso_suffix, 'ssw.c', 'ssw.h',
```

File: var/spack/repos/builtin/packages/isescan/package.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            )
+
+        # set paths to required programs
+        blast_pfx = self.spec['blast-plus'].prefix.bin
+        blastn_path = join_path(blast_pfx, 'blastn')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/isescan/package.py]:
```suggestion
        blastn_path = blast_pfx.blastn
```
Same for other lines

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by codeandkey** [comment]
Updated -- I switched the PIC flag and DSO suffix, but is there a generic alternative for `-rdynamic` and `-shared`?

**Comment by codeandkey** [comment]
Restarting for CI now that dependencies have been merged.

**Review by adamjstewart — APPROVED** [review]

**Comment by codeandkey** [comment]
Oops, forgot to push rebase. Should be safe to merge now if ready.

**Pull request merged**